src/integrations/database/implementations/mongodb.ts

```diff
@@ -0,0 +1,289 @@
+    
+        for (const key in multiStatements)
+        {
+            if (key === 'AND' || key ==='OR')
+            {
+                const singleMultiStatements  = multiStatements[key] ?? [];
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                const multiMongoQuery: Filter<any>[] = [];
+                
+                for (const statement of singleMultiStatements)
+                {
+                    const mongoQuery = this.#buildMongoQuery(statement);
+                    multiMongoQuery.push(mongoQuery);
+                }
+
+                const mongoKey = BOOLEAN[key];
+                mongoQuery[mongoKey] = multiMongoQuery;
+
+                continue;
+            }
+            
+            const expression = singleStatements[key];
+            const mongoKey = key === 'id' ? '_id' : key;
+            const mongoExpression: Record<string, unknown> = {};
+            
+            for (const operator in expression)
+            {
+                const value = this.#extractValue(expression as RecordData, operator as QueryOperator);
+                const mongoValue = key === 'id' ? this.#createId(value as string) : value;
+                const mongoOperator = OPERATORS[operator];
+                mongoExpression[mongoOperator] = mongoValue;
+            }
+
+            mongoQuery[mongoKey] = mongoExpression;
+        }   
+
+        return mongoQuery;
+    }
+
+    #buildMongoSort(sort?: RecordSort): Sort
+    {
+        const mongoSort: Sort = {};
+
+        if (sort === undefined)
+        {
+            return mongoSort;
+        }
+
+        for (const element in sort)
+        {
+            const direction = sort[element];
+            mongoSort[element] = direction === SortDirections.DESCENDING ? -1 : 1;
+        }
+
+        return mongoSort;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async #getCollection<T>(name : RecordType): Promise<Collection<T extends Document ? any: any>>
+    {
+        if (this.#database === undefined)
+        {
+            throw new NotConnected();
+        }
+
+        return this.#database.collection(name);
+    }
+
+    #getDatabase(databaseName: string): Db
+    {   
+        if (this.#client === undefined)
+        {
+            throw new NotConnected();
+        }
+        
+        return this.#client.db(databaseName);   
+    }
+
+    async #createClient(connectionString: string): Promise<MongoClient>
+    {
+        try
+        {
+            return await MongoClient.connect(connectionString);
+        }
+        catch (error: unknown)
+        {
+            const message = error instanceof Error ? error.message : undefined;
+
+            throw new NotConnected(message);
+        }
+    }
+
+    #buildRecordData(data: Document, fields?: RecordField[]) : RecordData
+    { 
+        let result: RecordData = {};
+
+        if (fields === undefined)
+        {
+            result = {...data};
```

Comment:
Inconsistent spacing for the given object(s).

Reply:
inconsistency solved.